utils/crypto.go

```diff
@@ -2,35 +2,64 @@ package utils
 
 import (
 	"crypto/ecdsa"
+	"encoding/binary"
 	"encoding/hex"
 	"fmt"
 	"github.com/ethereum/go-ethereum/crypto"
 	"github.com/juju/loggo"
 )
 
-
 var log = loggo.GetLogger("utils")
 
 func LaSign(data []byte, prv *ecdsa.PrivateKey, chainId byte) ([]byte, error) {
 
 	dataHash := crypto.Keccak256(data)
 	signature, err := crypto.Sign(dataHash, prv)
-
 	if err != nil {
 		return nil, err
 	}
-	signature[64] = chainId*2 + 35 + signature[64]
-	return signature, nil
+
+	if chainId < 110 {
+		signature[64] = chainId*2 + 35 + signature[64]
+		return signature, nil
+	}
+	result := make([]byte, 65)
+	encodedRecId := uint32(chainId)*2 + 35 + uint32(signature[64])
+	recIdBytes := make([]byte, 4)
+	binary.BigEndian.PutUint32(recIdBytes, encodedRecId)
```

Comment:
I think it should be binary.LittleEndian.PutUint32 if you wish to keep the lowest byte in lowest index. I checked with online go lang compiler. Can your confirm it please?

Reply:
yes,  you are right,  missed it.)